IGNITE-14417 Document performance-statistics-ext module (#8940)
apache/ignite: docs/_docs/extensions-and-integrations/performance-statistics.adoc

// Licensed to the Apache Software Foundation (ASF) under one or more
// contributor license agreements.  See the NOTICE file distributed with
// this work for additional information regarding copyright ownership.
// The ASF licenses this file to You under the Apache License, Version 2.0
// (the "License"); you may not use this file except in compliance with
// the License.  You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
= Performance Statistics Extension

== Overview

Ignite provides link:monitoring-metrics/performance-statistics[a built-in tool for cluster profiling.]

> The extension provides utilities to link:#building-the-report[create the report] and
> link:#print-statistics[print statistics] from collected files.

Reviewer: ```suggestion
The extension provides utilities to link:#building-the-report[create the report] and link:#print-statistics[print statistics] from the collected files.
```

== Building the Report

> Ignite provides a tool to generate the report from performance statistics files.

Reviewer: ```suggestion
Ignite provides a tool to generate the report from the performance statistics files.
```

Follow these steps to build the performance report:

1. Stop collecting statistics and place files from all nodes under an empty directory. For example:

    /path_to_files/
        ├── node-162c7147-fef8-4ea2-bd25-8653c41fc7fa.prf
        ├── node-7b8a7c5c-f3b7-46c3-90da-e66103c00001.prf
        └── node-faedc6c9-3542-4610-ae10-4ff7e0600000.prf

2.  Run the script from the release package of the tool:

>         performance-statistics-tool/build-report.sh path_to_files

Reviewer: ```suggestion
[tabs]
--
tab:Unix[]
[source,shell]
----
performance-statistics-tool/build-report.sh path_to_files
----
tab:Windows[]
[source,shell]
----
performance-statistics-tool/build-report.bat path_to_files
----
--
```

> The performance report is created in the new directory under the performance statistics files
> path: `path_to_files/report_yyyy-MM-dd_HH-mm-ss/`. Open `report_yyyy-MM-dd_HH-mm-ss/index.html` in the browser to see
> the report.

Reviewer: ```suggestion
The performance report is created in the new directory under the performance statistics files: `path_to_files/report_yyyy-MM-dd_HH-mm-ss/`. 
Open `report_yyyy-MM-dd_HH-mm-ss/index.html` in the browser to see the report.
```

For more details run the help command:

>     performance-statistics-tool/build-report.sh --help

Reviewer: ```suggestion
[tabs]
--
tab:Unix[]
[source,shell]
----
performance-statistics-tool/build-report.sh --help
----
tab:Windows[]
[source,shell]
----
performance-statistics-tool/build-report.bat --help
----
--  
```

> == Print statistics

Reviewer: ```suggestion
== Print Statistics
```

> Ignite provides a tool to print statistics to console or file in JSON format.

Reviewer: ```suggestion
Ignite provides a tool to print statistics to a console or to a file in JSON format.
```

Run the script from the release package of the tool to print statistics:

>     performance-statistics-tool/print-statistics.sh path_to_files

Reviewer: ```suggestion
[tabs]
--
tab:Unix[]
[source,shell]
----
performance-statistics-tool/print-statistics.sh --path_to_files
----
tab:Windows[]
[source,shell]
----
performance-statistics-tool/print-statistics.bat --path_to_files
----
--  
```

> where `path_to_files` is a path to performance statistics file or files directory.

Reviewer: ```suggestion
Note that `path_to_files` is a path to the performance statistics file or files directory.
```

The script provides the ability to filter operations by operation's type, time, or cache. For more details run the
help command:

>     performance-statistics-tool/print-statistics.sh --help

Reviewer: ```suggestion
[tabs]
--
tab:Unix[]
[source,shell]
----
performance-statistics-tool/print-statistics.sh --help
----
tab:Windows[]
[source,shell]
----
performance-statistics-tool/print-statistics.bat --help
----
-- 
```

> Example of output:

Reviewer: ```suggestion
See the output example below:
```
